cw_decred/lib/wallet_addresses.dart

```diff
@@ -26,20 +68,42 @@ class DecredWalletAddresses extends WalletAddresses {
     return nAddr;
   }
 
+  List<AddressInfo> getAddressInfos() {
+    if (addressInfos.containsKey(0)) {
+      return addressInfos[0]!;
+    }
+    return <AddressInfo>[];
+  }
+
   List<String> addresses() {
-    final res = libdcrwallet.addresses(walletInfo.name);
-    final addrs = (json.decode(res) as List<dynamic>).cast<String>();
+    final addrs = List<String>.from(usedAddresses);
+    for (final k in addressesMap.keys) {
+      addrs.add(k);
+    }
     return addrs;
   }
 
-  @override
-  set address(String addr) {}
-
-  @override
-  Future<void> init() async {}
-
-  @override
-  Future<void> updateAddressesInBox() async {
+  Future<void> updateAddress(String address, String label) async {
+    if (!addressInfos.containsKey(0)) {
+      return;
+    }
+    addressInfos[0]!.forEach((info) {
+      if (info.address == address) {
+        info.label = label;
+      }
+    });
     await saveAddressesInBox();
   }
+
+  (List<String>, List<String>, int) libAddresses() {
+    final nUsed = "10";
+    final nUnused = "1";
+    final res = libdcrwallet.addresses(walletInfo.name, nUsed, nUnused);
```

Comment:
In the libwallet go code, I saw this comment...
> No unused addresses are returned if nUnused is zero. All used addresses are returned if nUsed is zero.

Is there ever a time we need all used addresses? Should `nUsed` return NO used addresses, rather than ALL used addresses?

I also don't see a use case for `nUsed` being 0 in the dart code so far.

Reply:
Probably zero being none makes more sense tbh. The old code returned all used and no unused, so I guess I was trying to make it like that. We probalby won't use zero either way so you can change it some time if you are doing something else in libwallet.